src/components/tx/ApprovalEditor/index.tsx

```diff
@@ -29,17 +30,26 @@ const Title = () => {
   )
 }
 
-export const ApprovalEditor = ({ safeTransaction }: { safeTransaction: SafeTransaction | undefined }) => {
+export const ApprovalEditor = ({
+  safeTransaction,
+  safeMessage,
+}: {
+  safeTransaction?: SafeTransaction
+  safeMessage?: EIP712TypedData
+}) => {
```

Comment:
With `safeMessage` added here we have to extend a lot of the conditions and it propagates down through `useApprovalInfos` to the `ApprovalModule` and adds more complexity.

Afaiu if there is a `safeMessage` we are always in readOnly so we could split this into two components, one for transactions one for safe messages and add a new method `ApprovalModule.scanMessage` to handle this case and check inside `useApprovalInfos` in one place if `safeMessage ? ApprovalModule.scanMessage : ApprovalModule.scanTransaction`. Wdyt?

Reply:
I think adding a scanMessage function helps readability of the `ApprovalModule` but I do not think we need to create a separate component as most of the components logic is shared / we use the same type for Permits as for Approvals.

I will refactor it tomorrow a bit more :)